hw/ip/otp_ctrl/doc/_index.md

```diff
@@ -25,30 +301,495 @@ The following table lists the main parameters used throughout the OTP controller
+For this the design at hand, it is assumed that this native OTP word-width is 16bit.
+For software convenience, however, these details are abstracted and the open-source OTP controller exposes the OTP storage as a linear address space of 32bit words, which is aligned with the machine word size of the Ibex processor.
+Since the OTP IP employs a redundancy mechanism similar to ECC, this implies however that write operations take place at a granularity of 32bit blocks for non-secret and 64bit blocks for secret partitions (due to the scrambling).
+Hence, software is responsible to appropriately pack and program items, since each 32bit location can only be programmed once.
+
+#### Life Cycle View
+
+Since the life cycle partition is the only partition that needs live updates in-field, proper care must be taken to properly encode data in this partition such that incremental updates are possible.
+The life cycle state is hence encoded such that incremental updates to the state are always carried out at the granularity of a 16bit word.
+Further, the life cycle transition counter is encoded such that each stroke consumes a full 16bit word for the same reason.
+
+See [**TODO: add link to lifecycle spec and state encoding**] for more details on the life cycle encoding.
+
+### Partition Implementations
+
+There are three different partition types that are reused throughout the implementation:
+
+1. Unbuffered partition
+2. Buffered partition
+3. Life cycle partition
+
+Each partition type has a different RTL implementation, but what they have in common is that each partition has its own controller and set of local buffer registers (if required).
+
+#### Unbuffered Partition
+
+**TODO: add FSM details**
+
+#### Buffered Partition
+
+**TODO: add FSM details**
+
+#### Life-Cycle Partition
+
+**TODO: add FSM details**
+
+### Scrambling Datapath
+
+![OTP Digest Mechanism](otp_digest_mechanism.svg)
+
+The scrambling datapath is built around a single-round implementation of the [PRESENT lightweight cipher]({{< relref "hw/ip/prim/doc/prim_present" >}}) and contains some additional multiplexing circuitry to enable multiple modes of operation that are explained in more detail further below.
+
+#### Scrambling Mode
+
+As illustrated in subfigure a) in the diagram above, the standard 128bit-key PRESENT configuration with 31 rounds is used for scrambling operations.
+The key used for scrambling is a global netlist constant chosen by the silicon creator, and all secret partitions are encrypted using the same netlist constant.
+Note that the amount of data that is being scrambled is small (160byte = 20 x 64bit blocks) and the scrambled data remains constant.
+Hence, no additional masking or diversification scheme is applied since only a very limited amount of information can be gathered by observing the scrambling operation via side-channels.
+
+**TBD: should each secret partition have its own scrambling key constant?**
```

Comment:
this sounds like a good idea.

Reply:
ok, I'll take note and add a sentence describing this.